### PR TITLE
[AutoDiff] [Docs] Clarify 'Differentiable' derived conformances conditions.

### PR DESCRIPTION
**Pull request by rxwei**
Clarify `Differentiable` derived conformances synthesis conditions.

### CODE REVIEW

**Review by dan-zheng — APPROVED** [review]

**Comment by rxwei** [comment]
@swift-ci please smoke test and merge

**Comment by rxwei** [comment]
@swift-ci please smoke test and merge

**Comment by rxwei** [comment]
@swift-ci please smoke test and merge

**Comment by rxwei** [comment]
@swift-ci please smoke test and merge

**Comment by rxwei** [comment]
@swift-ci please smoke test and merge

File: docs/DifferentiableProgramming.md [hunk 1]
```diff
@@ -1302,7 +1389,7 @@ Method `move(along:)` will not be synthesized because a default implementation
 already exists.
 
 ```swift
-struct Point<T: Real>: @memberwise Differentiable, @memberwise AdditiveArithmetic {
+struct Point<T: Real>: @memberwise Differentiable, AdditiveArithmetic {
```
Comment (dan-zheng) [inline docs/DifferentiableProgramming.md:1392]: Would the `@memberwise` attribute apply to both the `Differentiable` conformance and the `AdditiveArithmetic` conformance?

File: docs/DifferentiableProgramming.md [hunk 2]
```diff
@@ -1289,6 +1308,74 @@ struct Foo<T: Differentiable, U: Differentiable>: @memberwise Differentiable {
...
+}
+```
+
+For clarity as to which stored properties are to be included for
+differentiation, the compiler will recommend that all stored properties that
+cannot be included as differentiable variables (due to either lacking a
+conformance to `Differentiable` or being a non-`class`-bound `let` property) be
+marked with `@noDerivative`. When a property is not included as a differentiable
+variable and is not marked with `@noDerivative`, the compiler produces a warning
+as asking the user to make the exclusion explicit along with fix-it suggestions
```
Comment (dan-zheng) [inline docs/DifferentiableProgramming.md:1343]:
```suggestion
asking the user to make the exclusion explicit along with fix-it suggestions
```

File: docs/DifferentiableProgramming.md (hunk 1 above)
Comment (rxwei) [reply docs/DifferentiableProgramming.md]: This change was unintended. Fixed.

File: docs/DifferentiableProgramming.md (hunk 2 above)
Comment (rxwei) [reply docs/DifferentiableProgramming.md]: Thanks for the catch. Fixed.

**Comment by rxwei** [comment]
@swift-ci please smoke test and merge

**Comment by rxwei** [comment]
@swift-ci please smoke test and merge

**Pull request merged**